From 652864c2515fa3a8e8589b26220a62b3bfd6fe59 Mon Sep 17 00:00:00 2001
From: Chetan Rao <chetan.rao@intel.com>
Date: Fri, 24 May 2013 10:42:42 +0300
Subject: build: New macro to extended the AOSP include directories

New macro to extended the AOSP include directories

Bug: 104519
Category: AOSP Improvement
Domain: AOSP-build
Source: Internal
Upstream-candidate: Yes

Change-Id: I324ec4d4ab354f20e8be81d67352046ebb9ce831
Signed-off-by: Chetan Rao <chetan.rao@intel.com>
Reviewed-on: http://android.intel.com:8080/109700
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Rocard, KevinX <kevinx.rocard@intel.com>
Reviewed-by: Gonzalve, Sebastien <sebastien.gonzalve@intel.com>
Reviewed-by: Fert, Laurent <laurent.fert@intel.com>
Reviewed-by: Wu, James Y <james.y.wu@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 core/pathmap.mk |   23 ++++++++++++++++++++++-
 1 files changed, 22 insertions(+), 1 deletions(-)

diff --git a/core/pathmap.mk b/core/pathmap.mk
index 5dfc543..5dbb7a4 100644
--- a/core/pathmap.mk
+++ b/core/pathmap.mk
@@ -57,7 +57,8 @@ pathmap_INCL := \
 #
 # Returns the path to the requested module's include directory,
 # relative to the root of the source tree.  Does not handle external
-# modules.
+# modules by default. external modules can be supported by using
+# add-path-map macro in device/intel/common/BroadConfig.mk.
 #
 # $(1): a list of modules (or other named entities) to find the includes for
 #
@@ -66,6 +67,26 @@ $(foreach n,$(1),$(patsubst $(n):%,%,$(filter $(n):%,$(pathmap_INCL))))
 endef
 
 #
+# Macro to add include directories of modules in pathmap_INCL
+# relative to root of source tree. Usage:
+# $(call add-path-map, project1:path1)
+# OR
+# $(call add-path-map, \
+#        project1:path1 \
+#        project2:path1)
+#
+define add-path-map
+$(eval pathmap_INCL += \
+    $(foreach path, $(1), \
+        $(if $(filter $(firstword $(subst :, ,$(path))):%, $(pathmap_INCL)), \
+            $(error Duplicate AOSP path map $(path)), \
+            $(path) \
+         ) \
+     ) \
+ )
+endef
+
+#
 # Many modules expect to be able to say "#include <jni.h>",
 # so make it easy for them to find the correct path.
 #
-- 
1.7.4.1

